src/Bicep.Core/SemanticModel/NameBindingVisitor.cs

```diff
@@ -121,19 +146,44 @@ private Symbol LookupSymbolByName(string name, TextSpan span)
             // symbol does not exist in the local namespace
             // try it in the imported namespaces
 
-            // match in one of the namespaces
-            var foundSymbols = this.namespaces
-                .Select(ns => ns.TryGetFunctionSymbol(name))
-                .Where(symbol => symbol != null)
-                .ToList();
+            // check if name is a namespace symbol first
+            var namespaceSymbol = this.namespaces
+                .Where(ns => ns.Name.Equals(name, System.StringComparison.OrdinalIgnoreCase))
```

Comment:
>System.StringComparison.OrdinalIgnoreCase [](start = 50, length = 41)

Identifier comparisons are supposed to be case-sensitive. In `LanguageConstants` we have an IdentifierComparer. Can you use that instead? (It's the same as the Ordinal one, but it also acts as a way to find all the places where identifiers are compared.)

Reply:
Code updated and moved namespace lookup as a local function.